cypress/integration/control_data.js

```diff
@@ -0,0 +1,110 @@
+context('Data Control', () => {
+	before(() => {
+		cy.login();
+		cy.visit('/app/doctype');
+		return cy.window().its('frappe').then(frappe => {
+			return frappe.xcall('frappe.tests.ui_test_helpers.create_doctype', {
+				name: 'Test Data Control',
+				fields: [
+					{
+						"label": "Name",
+						"fieldname": "name1",
+						"fieldtype": "Data",
+						"options": "Name",
+						"in_list_view": 1,
+					},
+					{
+						"label": "Email-ID",
+						"fieldname": "email",
+						"fieldtype": "Data",
+						"options": "Email",
+						"in_list_view": 1,
+					},
+					{
+						"label": "Phone No.",
+						"fieldname": "phone",
+						"fieldtype": "Data",
+						"options": "Phone",
+						"in_list_view": 1,
+					},
+				]
+			});
+		});
+	});
+	it('Verifying data control by inputting different patterns', () => {
+		cy.new_form('Test Data Control');
+
+		//Checking the URL for the new form of the doctype
+		cy.location("pathname").should('eq', '/app/test-data-control/new-test-data-control-1');
+		cy.get('.title-text').should('have.text', 'New Test Data Control');
+
+		//Checking if the status is "Not Saved" initially
+		cy.get('.indicator-pill').should('have.text', 'Not Saved');
+
+		//Inputting data in the field
+		cy.fill_field('name1', '@@###', 'Data');
+
+		//Checking if the border color of the field changes to red
+		cy.get_field('name1', 'Data').should('have.css', 'border', '1px solid rgb(236, 100, 94)');
+		cy.findByRole('button', {name: 'Save'}).click();
+
+		//Checking for the error message
+		cy.get('.modal-title').should('have.text', 'Message');
+		cy.get('.msgprint').should('have.text', '@@### is not a valid Name');
+		cy.reload();
+
+		cy.fill_field('name1', 'Komal{}/!', 'Data');
+		cy.get_field('name1', 'Data').should('have.css', 'border', '1px solid rgb(236, 100, 94)');
```

Comment:
Instead of checking for absolute color, let's check whether the `has-error` class is applied properly and `redq` class is set to the label of the field.

Reply:
Done